source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -87,15 +87,49 @@ void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callb
   callbacks_ = &callbacks;
 }
 
+Http::FilterHeadersStatus Filter::encode100ContinueHeaders(Http::HeaderMap&) {
+  return Http::FilterHeadersStatus::Continue;
+}
+
+Http::FilterHeadersStatus Filter::encodeHeaders(Http::HeaderMap& headers, bool) {
+  if (headers_to_add_) {
+    headers_to_add_->iterate(
+        [](const Http::HeaderEntry& header, void* context) -> Http::HeaderMap::Iterate {
```

Comment:
I removed the duplication in ratelimit filter.
I couldn't find a method in HeaderMap that allows me to add another HeaderMap.
Short of adding such method, not sure if there is a way to avoid duplication between HeaderMapImpl and ratelimit.cc iterations?

Also any idea if I can "move" elements from one HeaderMap to another without doing the extra copy via HeaderString?